pkg/shell/machines/machines.js

```diff
@@ -379,6 +379,8 @@ function Machines() {
         let port_spot = -1;
 
         if (conn_to) {
+            if (conn_to.substring(0, 6) === "ssh://")
+                conn_to = conn_to.substring(6);
```

Comment:
I actually wonder where that `ssh://` thing even  comes from. https://git.libssh.org/projects/libssh.git/ has no idea about it. Commit bb7a8b1ac3 just introduced it into the help en passant without any references to documentation or tests. So I think it's pretty much up to us to define it. As the only plausible format is `ssh://[user@]host[:port]`, and we already support `[user@]host[:port]`, stripping the prefix seems fine to me.

So the implementation is good, but we need some integration test in test/verify/check-shell-host-switching or test/verify/check-shell-multi-machine to ensure that `ssh://host`, ssh://host:port`, and `ssh://user@host:port` all work.

Reply:
Hello @martinpitt,
I've added a test for the ssh connections (and modified `add_machine` to get the username). I wasn't sure where to put this exactly so I put it in a new test case `testSSH` for now. Let me know if this is correct

Reply:
Thanks! I'll comment on the new test in separate threads. The `machine` argument is indeed unused. Feel free to add a separate commit to remove it from the declaration and all of its callers.

Please also squash the new test into the first commit, and prefix the subject with "shell: ". Finally, please rebase your branch to latest origin/main so that the packit tests keep working (it's a good habit to `git fetch origin; git rebase origin/main` before *every* PR push).